pandas/core/indexes/multi.py

```diff
@@ -50,6 +50,85 @@
          target_klass='MultiIndex or list of tuples'))
 
 
+class MultiIndexUIntEngine(libindex.BaseMultiIndexCodesEngine,
```

Comment:
is there any reason why these are defined in python, rather than in cython? these could then be cdef classes, and you can type _codes_to_int.

Reply:
cython doesn't allow multiple inheritance (and avoiding it would complicate the code a lot)